test/end2end_test.go

```diff
@@ -3901,6 +3901,36 @@ func (s) TestClientRequestBodyErrorCloseAfterLength(t *testing.T) {
 	}
 }
 
+func testClientInvalidStreamID(t *testing.T, e env) {
+	te := newTest(t, e)
+	ts := &funcServer{streamingInputCall: func(stream testgrpc.TestService_StreamingInputCallServer) error {
+		_, err := stream.Recv()
+		return err
+	}}
+	te.startServer(ts)
+	defer te.tearDown()
+	te.withServerTester(func(st *serverTester) {
+		st.writeHeadersGRPC(2, "/grpc.testing.TestService/StreamingInputCall", true)
+		_, err := st.fr.ReadFrame()
+		if err == nil {
```

Comment:
We should assert for the exact error here. would be hard to catch the situation where ReadFrame could have error'd for an unrelated issue

Reply:
I was looking at how to assert the error here. 

```
--- FAIL: Test (0.00s)
    --- FAIL: Test/ClientInvalidStreamID (0.00s)
        end2end_test.go:3917: Running test in tcp-clear-v1-balancer environment...
        tlogger.go:116: INFO server.go:667 [core] [Server #1] Server created  (t=+1.368417ms)
        tlogger.go:116: INFO server.go:867 [core] [Server #1 ListenSocket #2] ListenSocket created  (t=+1.448209ms)
        tlogger.go:116: INFO syscall_nonlinux.go:39 [core] CPU time info is unavailable on non-linux environments.  (t=+2.208959ms)
        tlogger.go:116: INFO http2_server.go:1210 [transport] [server-transport 0x140002a89c0] Closing: received an illegal stream id: 2. headers frame: [FrameHeader HEADERS flags=END_STREAM|END_HEADERS stream=2 len=60]  (t=+2.488042ms)
        tlogger.go:116: INFO controlbuf.go:543 [transport] [server-transport 0x140002a89c0] loopyWriter exiting with error: transport closed by client  (t=+2.506959ms)
        end2end_test.go:3928: Error : EOF
        tlogger.go:116: INFO server.go:806 [core] [Server #1 ListenSocket #2] ListenSocket deleted  (t=+2.577042ms)
FAIL
```

This is the error log for me.  I was printing what error I was getting and it is EOF. It makes sense because there are no frames sent from the server. Can you suggest how to verify the expected error? string matching :p?

Reply:
One more thought, Instead of the test here, can we do a test in the `transport.go` and test http2_server's operateHeaders func. let me see